frappe/database/db_manager.py

```diff
@@ -79,7 +79,7 @@ def restore_database(verbose: bool, target: str, source: str, user: str, passwor
 			command.extend(["cat", source, "|"])
 
 		# Newer versions of MariaDB add in a line that'll break on older versions, so remove it
-		command.extend(["sed", r"'/\/\*!999999\\- enable the sandbox mode \*\//d'", "|"])
+		command.extend(["sed", r"'/\/\*M\{0,1\}!999999\\- enable the sandbox mode \*\//d'", "|"])
```

Comment:
Looks like this should've been `sed -E`

Reply:
I believe not, sed without `-E` should support the `{x,y}` quantifier (but not support the `?` quantifier)

https://stackoverflow.com/a/6157705

Reply:
Maybe it's the extra backslash `\` before the brackets `{` 😅😅

Reply:
The generated command doesn't include that - will share in a few minutes

Reply:
```bash
sed '/\/\*M{0,1}!999999\\- enable the sandbox mode \*\//d'```
```

Reply:
```bash
❯ gzip -cd 20240920_132344-v16_localhost-database.sql.gz | sed '/\/\*M{0,1}!999999\\- enable the sandbox mode \*\//d' | head -8
-- begin frappe metadata
-- [frappe]
-- version = 16.0.0-dev
-- branch = refactor-role-settings
-- end frappe metadata
-- 
/*M!999999\- enable the sandbox mode */ 
-- MariaDB dump 10.19-11.5.2-MariaDB, for Linux (x86_64)
❯ gzip -cd 20240920_132344-v16_localhost-database.sql.gz | sed -E '/\/\*M{0,1}!999999\\- enable the sandbox mode \*\//d' | head -8
-- begin frappe metadata
-- [frappe]
-- version = 16.0.0-dev
-- branch = refactor-role-settings
-- end frappe metadata
-- 
-- MariaDB dump 10.19-11.5.2-MariaDB, for Linux (x86_64)
--

```

Reply:
Okay it's the backslashes, they are missing I think

```diff
- sed '/\/\*M{0,1}!999999\\- enable the sandbox mode \*\//d'
+ sed '/\/\*M\{0,1\}!999999\\- enable the sandbox mode \*\//d'
```

![](https://github.com/user-attachments/assets/cf928a8b-921e-45ec-b6a7-df916f280969)

Reply:
Because of shell expanding?

Reply:
Let's just use the `-E` flag with the `?` quantifier?

Reply:
hmm @cogk looks like my local v16 code was a bit messed up, seems fine now :cry: 

(I began going through this and testing when the v14 backport didn't seem to work, let me check what is happening there...)